### ISSUE TITLE
Email quotas reached out

### ISSUE THREAD

**Issue by filippomc** [OpenSourceBrain/OSBv2#582]

**Comment by filippomc** [comment]
investigate possible solutions for now, longer term the fix

**Comment by zsinnema** [comment]
@filippomc I've installed a local smtp server in both OSB clusters (dev and prod)
+ configured keycloak to use the local smtp server
+ created codefresh pipelines to install the smtp relay servers (in the osb project)

**Comment by sanjayankur31** [comment]
Could you send out some test e-mails so we can check if all the SPF settings etc. are correct (ie., so that these don't end up being marked as spam?)

**Comment by zsinnema** [comment]
@sanjayankur31 we can test it by sending an email through the keycloak admin.
I tried it and the relay server log shows that the credentials for smtp.ionos.co.uk are invalid

do you have the right credentials for me?

**Comment by zsinnema** [comment]
Hi @sanjayankur31 I got the smtp relay details from @filippomc

**Comment by zsinnema** [comment]
@sanjayankur31 you can test the email by logging into the keycloak server and go to the mail settings: https://accounts.v2.opensourcebrain.org/auth/admin/master/console/#/realms/osb2/smtp-settings
on the right there is a "test connection" button. This button will send an email to your email address.

**Comment by sanjayankur31** [comment]
Tested, seems to work well. SPF etc. correctly set.

**Comment by sanjayankur31** [comment]
Question: since this is still using our Ionos account, will they not still block the account etc. when lots of e-mails are sent out in a short period of time? Does running our own smtp server by pass that? Or is this something we now need to test with this local smtp server solution?

**Comment by zsinnema** [comment]
@sanjayankur31 yes it still uses your Ionos account but now we can rate limit the sending of emails. currently it's sending 1 mail per 5s. Please let me know if you would like to change this setting

**Comment by sanjayankur31** [comment]
No, that's fine. I mean unless we know what the appropriate rate is, we can't really optimise this.

From a user perspective, this means that there will be some lag between users registering and receiving the confirmation e-mail. If this is going to be a noticeable lag, do we we need to tweak the registration pages to say something like "please check your e-mail for the confirmation e-mail, this may take upto N minutes to arrive" ?

**Comment by zsinnema** [comment]
with the above mentioned setting it will rate limit to one email message every 5 seconds. WHen there are a lot of users registering at once, let's say 36... then the last email will be send after 36 / one per 5s = 36/12 = 3 minutes
@sanjayankur31 do you know the rate limit at Ionos for sending mail?

**Comment by sanjayankur31** [comment]
No, we don't know the rate limit. I think Padraig had looked into this but wasn't able to find any info on it. Seems like it isn't documented anywhere.

I think the UCL course we're hoping to use OSBv2 has about 40 people, so yeh, that's a good approximation. The next time we do online tutorials, we'll hopefully have more, but we can work on that when we have some more data/numbers.

**Comment by filippomc** [comment]
https://www.ionos.com/help/email/general-topics/what-are-the-mass-mailing-limits/ 10 mails per minute seems to be the relevant limit here

**Comment by pgleeson** [comment]
Ok, so changing it to send one mail every say 8 seconds should be fine. We can tweak it if there are problems in future.

**Comment by zsinnema** [comment]
@pgleeson I put it to 1 mail per 8s
